src/read.c

```diff
@@ -2555,11 +2551,7 @@ ExecStatus ReadEvalCommand(Obj context, Obj *evalResult, UInt *dualSemicolon)
     STATE(Prompt) = SyQuiet ? "" : "> ";
 
     /* remember the old reader context                                     */
-    stackNams   = rs->StackNams;
-    readTop     = rs->ReadTop;
-    readTilde   = rs->ReadTilde;
     tilde       = STATE(Tilde);
-    currLHSGVar = rs->CurrLHSGVar;
```

Comment:
Here something "real" happens: we avoid the need to backing up parts of the ReaderState (and we also avoid questions about why we just save *those* parts of the Reader, but not others... Now we "save" all of them, automatically.)